src/libraries/filcrypto/filproofs/algorithms.go

```diff
@@ -26,16 +26,67 @@ type PieceInfo = sector.PieceInfo
 type Label Bytes32
 type Commitment = sector.Commitment
 
+type PrivatePostCandidateProof = sector.PrivatePoStCandidateProof
+
 const WRAPPER_LAYER_WINDOW_INDEX = -1
 
-func WinSDRParams(cfg SDRCfg) *WinStackedDRG_I {
+const NODE_SIZE = 32
+const ELECTION_POST_PARTITIONS = 1
+const SURPRISE_POST_PARTITIONS = 1
+const POST_LEAF_CHALLENGE_COUNT = 66
+const POST_CHALLENGE_RANGE_SIZE = 1
```

Comment:
some/all of these constants seem to be part of a single configuration, they should probably be bundled as such

Reply:
> I think this definitely improves things 👍
> 
> Im not sure, but i think this is still very hard to change to add a new type of proof protocol, especially something that differs substantially.
> 
> I would have expected parametrization to yield different types to exist for each of the relevant algorithms, and parametrizing function calls to the functions common to all algorithms. the current approach explodes out details of the algorithm all over, which puts type switches in a bunch of critical sections to use one detail or another. it is not clear if a substantially different algorithm can be added easily.
 
I avoided Go interfaces in favor of `.id` unions because I thought significant data structure were required to be defined in `.id`. If idiomatic Go is a viable solution, I agree what you suggest would be better, and we can make that change later if we decide it's worthwhile.

As discussed elsewhere, I'm going to merge this now in order to facilitate other refactoring and spec work. We can clean up loose ends then.